packages/template-retail-react-app/app/pages/checkout/partials/contact-info.jsx

```diff
@@ -72,11 +72,14 @@ const ContactInfo = () => {
             } else {
                 await login.mutateAsync({username: data.email, password: data.password})
 
-                // Because we lazy load the basket there is no guarantee that a basket exists for the newly registered
-                // user, for this reason we must transfer the ownership of the previous basket to the logged in user.
-                await transferBasket.mutateAsync({
-                    parameters: {overrideExisting: true}
-                })
+                const hasBasketItem = basket.productItems?.length > 0
+                if (hasBasketItem) {
+                    mergeBasket.mutate({
+                        parameters: {
+                            createDestinationBasket: true
+                        }
+                    })
+                }
```

Comment:
As for the comment, I think we can remove it since if the destination basket does not exist on merge, we create it using the `createDestinationBasket` parameter.